code/parachain/runtime/common/src/ibc.rs

```diff
@@ -31,3 +38,127 @@ where
 }
 
 pub type MinimumConnectionDelaySeconds = ConstU64<10>;
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum MemoMiddlewareNamespaceChain {
+	Forward { next: Option<Box<Self>> },
+	Wasm { next: Option<Box<Self>> },
```

Comment:
prevents forwarding evm memo

Reply:
any non recognizable attribute is ok.

Reply:
memo can be empty json and other fields. and even  on json

https://github.com/cosmos/ibc/tree/main/spec/app/ics-020-fungible-token-transfer#using-the-memo-field

and still valid.

Reply:
In memo validation function just return if the memo is empty

Reply:
If you need to forward evm memo, just add it here

Reply:
i do no see test proving that non optional enum will return will no not error on empy json. also there is parser code on ibc repo which may fail. and also whole cvm, cw works with memo inside xc-core, which may be used by pallet xcm ibc hops. so this memo must be removed and be just string. anyway, cw and cvm was cancelled or on hold. i second time repeated my suggestion. so if not listening, pleae fill free whatever way it will work according memo spec in ics20 at least. which by spec supports any attributes, i mentioned it 3 times already

Reply:
i do no see test proving that non optional enum will return will no not error on empy json. also there is parser code on ibc repo which may fail. and also whole cvm, cw works with memo inside xc-core, which may be used by pallet xcm ibc hops. so this memo must be removed and be just string. anyway, cw and cvm was cancelled or on hold. i second time repeated my suggestion. so if not listening, pleae fill free whatever way it will work according memo spec in ics20 at least. which by spec supports any attributes, i mentioned it 3 times already